src/Silex/Provider/MonologServiceProvider.php

```diff
@@ -77,14 +98,19 @@ public function register(Container $app)
             return new LogListener($app['logger'], $app['monolog.exception.logger_filter']);
         };
 
-        $app['monolog.name'] = 'myapp';
+        $app['monolog.name'] = 'app';
         $app['monolog.bubble'] = true;
         $app['monolog.permission'] = null;
         $app['monolog.exception.logger_filter'] = null;
+        $app['monolog.logfile'] = null;
     }
 
     public function boot(Application $app)
     {
+        if (!$app['debug']) {
+            ErrorHandler::register($app['monolog']);
+        }
```

Comment:
Do you really mean to only enable this in non-debug mode? I am not familiar enough with silex to know how it behaves internally in debug mode so I'm not sure.